### PR TITLE
Adding End-to-End Tests for Collection Creation Flow

### PR DESCRIPTION
**Pull request by magecnion**
## PR Type:
Tests

___
## PR Description:
This PR introduces end-to-end tests for the collection creation flow in the LaosEvolution contract. It includes the addition of a new contract file, updates to test configurations, and the creation of a new test file specifically for testing the collection creation process. The PR also modifies the build workflow to include a build step before running the tests.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`ownership-chain/e2e-tests/contracts/LaosEvolution.sol`: Introduced the LaosEvolution contract interface with methods for creating collections, minting tokens, and updating token metadata.
`ownership-chain/e2e-tests/tests/config.ts`: Added genesis account details to the test configuration.
`ownership-chain/e2e-tests/tests/test-create-collection.ts`: Created a new test suite for testing the collection creation process in the LaosEvolution contract. It includes tests for checking the genesis balance setup, verifying the owner of a newly created collection, and validating the emitted event upon collection creation.
`ownership-chain/e2e-tests/tests/util.ts`: Updated the describeWithExistingNode function to include ethersjs in the context.
`.github/workflows/build.yml`: Modified the test run command to include a build step before executing the tests.
</details>

### CODE REVIEW

**Review by dastanfv — COMMENTED** [review]
should we test failure scenarios?

File: ownership-chain/e2e-tests/tests/test-create-collection.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+describeWithExistingNode("Frontier RPC (Create Collection)", (context) => {
+    const LAOS_EVOLUTION_ABI = LaosEvolution.abi as AbiItem[]
+    const collectionId = "0";
+    const contractAddress = "0x0000000000000000000000000000000000000403";
+    const contract = new context.web3.eth.Contract(LAOS_EVOLUTION_ABI, contractAddress, {
+        from: GENESIS_ACCOUNT,
+        gasPrice: "0x3B9ACA00",
+    });
+
+    step("genesis balance is setup correctly", async function () {
```
Comment (asiniscalchi) [inline ownership-chain/e2e-tests/tests/test-create-collection.ts:17]: could we move it to the file containing generic tests ?

File: ownership-chain/e2e-tests/tests/util.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export async function customRequest(web3: Web3, method: string, params: any[]) {
 	});
 }
 
-export function describeWithExistingNode(title: string, cb: (context: { web3: Web3 }) => void, provider?: string) {
+export function describeWithExistingNode(title: string, cb: (context: { web3: Web3, ethersjs: ethers.JsonRpcProvider }) => void, provider?: string) {
```
Comment (asiniscalchi) [inline ownership-chain/e2e-tests/tests/util.ts:34]: is ti possible to avoid ethersjs ?

**Review by asiniscalchi — CHANGES_REQUESTED** [review]

**Comment by magecnion** [comment]
> should we test failure scenarios?



> should we test failure scenarios?

I would argue that we don't need them, as they are adequately tested in unit tests. I suggest utilizing end-to-end (e2e) tests to verify more critical functionalities such as events, since they serve as the source of truth for third parties. Given that e2e tests are more costly, and don't provide additional value when testing all failures, I believe focusing on critical functionalities is a more efficient approach. However, I am open to other perspectives.

**Review by asiniscalchi — APPROVED** [review]

**Pull request merged**